### ISSUE TITLE
no_grad for rollouts

### ISSUE THREAD

**Issue by danielhomola** [sid-sundrani/ppo_lightning#5]
I know you've followed the spinning up ppo implementation closely when implementing yours (and they  [don't seem to do this either](https://github.com/openai/spinningup/blob/master/spinup/algos/pytorch/ppo/ppo.py#L213)), but wouldn't it make sense to wrap [this line](https://github.com/sid-sundrani/ppo_lightning/blob/master/ppo_model.py#L173) into a
`with torch.no_grad():` clause? 

Why do we keep track of the gradients for the rollouts? I thought we only care about those for the learning with gradient descent bit.. What do you reckon?

**Comment by sidhantls** [comment]
In the line you're referring to, the call to `self.agent` is already wrapped in `no_grad`. See https://github.com/sid-sundrani/ppo_lightning/blob/fd9f17e30fff43f709936695cd3b731a5ad01704/networks.py#L109 

There's no need to keep track of gradients in the rollout

**Comment by danielhomola** [comment]
Ah, I missed that (again). Brilliant, thanks! I genuinely can't find it though where are they doing the same..

**Comment by sidhantls** [comment]
You're welcome. I briefly looked at the code snippet you linked- they call `ac.step` during the rollout, and they indeed use `torch.no_grad()` in the `step` method :)

**Comment by danielhomola** [comment]
ah good catch.. I'm blind basically :D
